### ISSUE TITLE
[a11y]: SelectableTile component has critical violation "The tabbable element's role 'none' is not a widget role"

### ISSUE THREAD

**Issue by kenneth-nguyen1** [carbon-design-system/carbon#12199]
### Package

carbon-components-react

### Browser

Chrome

### Operating System

MacOS

### Package version

7.56.0

### React version

17.0.2

### Automated testing tool and ruleset

IBM Equal Access Accessibility Checker - Latest Deployment

### Description

The SelectableTile component has a critical a11y violation "The tabbable element's role 'none' is not a widget role" that was found in our product using the IBM Accessibility Checker tool and is re-producible on the [v10](https://v7-react.carbondesignsystem.com/?path=/story/components-tile--multi-select) and [v11](https://react.carbondesignsystem.com/?path=/story/components-tile--multi-select) storybook as well

![SelectableTile-a11y](https://user-images.githubusercontent.com/114688923/193101544-6cac94f1-9b99-41db-a8df-e366240ca55b.png)

### Reproduction/example

Storybook v10 - https://v7-react.carbondesignsystem.com/?path=/story/components-tile--multi-select or Storybook v11 - https://react.carbondesignsystem.com/?path=/story/components-tile--multi-select

### Steps to reproduce

On the storybook link (either v10 or v11) for the Tile -> Multi Select story, run the IBM Accessibility Checker scan. The scan should report the violation mentioned above.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by tay1orjones** [comment]
This same error impacted StructuredList and required that we overhaul StructuredList. These changes included breaking changes, and I think the same thing could apply here.

We should validate this to make sure we can't fix this without a breaking change.

**Comment by kenneth-nguyen1** [comment]
@tay1orjones @tw15egan Can this change also be backported to carbon v10? That would be much appreciated!
